src/capabilities.rs

```diff
@@ -679,6 +690,27 @@ pub fn capabilities(meta: EditorMeta, ctx: &mut Context) {
             .entry("lsp-diagnostics".to_string())
             .or_default()
             .push(server_name);
+        probe_feature(to_editor, entry, &mut features, CAPABILITY_DIAGNOSTIC);
+
+        // Check for workspace diagnostics support
+        if server_has_capability(to_editor, server_settings, CAPABILITY_DIAGNOSTIC) {
+            if let Some(ref caps) = server_settings.capabilities {
+                let supports_workspace = match &caps.diagnostic_provider {
+                    Some(DiagnosticServerCapabilities::Options(opts)) => opts.workspace_diagnostics,
+                    Some(DiagnosticServerCapabilities::RegistrationOptions(opts)) => {
+                        opts.diagnostic_options.workspace_diagnostics
```

Comment:
we're sort of duplicating code here (i.e. we check `opts.diagnostic_options.workspace_diagnostics` in two places);
how about defining a `CAPABILITY_WORKSPACE_DIAGNOSTICS = "lsp-workspace-diagnostics"`,
and use that in both places?